src/decisionengine/framework/dataspace/maintain.py

```diff
@@ -171,9 +171,12 @@ def stop(self):
         """
         if isinstance(self.thread, threading.Thread):
             if self.thread.is_alive():
-                self.logger.debug("Sending reaper State.SHUTTINGDOWN signal")
                 with self.state.lock:
-                    self.state.set(State.SHUTTINGDOWN)
+                    if not self.state.should_stop():
+                        self.logger.debug("Sending reaper State.SHUTTINGDOWN signal")
+                        self.state.set(State.SHUTTINGDOWN)
+                    else:
+                        self.logger.debug(f"Reaper in state {self.state.get()} no need to call shutdown again")
```

Comment:
I'm confused, Pat.  As far as I understand, calling `self.state.set(State.SHUTTINGDOWN)` does not call shutdown, nor does it send a signal, per se.  It just ensures that the state of the reaper is the shutting-down state, regardless of its previous state (even if that state was `State.SHUTTINGDOWN`.  I'm therefore seeing no observable change in functionality here, but just more typing (and some potentially different logged messages).  What am I missing?

Reply:
I'm trying to unwravel : https://github.com/HEPCloud/decisionengine/pull/424/checks?check_run_id=2954255157#step:10:299

The logs seem to indicate that shutdown is running multiple times.... I may have missed what is actually going on there.

Reply:
As I think it through, there is a cleaner way to do this....

Reply:
I see, I'll take a look.